pkg/providers/instancetype/suite_test.go

```diff
@@ -759,13 +759,68 @@ var _ = Describe("InstanceType Provider", func() {
+				ExpectScheduled(ctx, env.Client, pod)
+
+				Expect(azureEnv.VirtualMachinesAPI.VirtualMachineCreateOrUpdateBehavior.CalledWithInput.Len()).To(Equal(1))
+				vm := azureEnv.VirtualMachinesAPI.VirtualMachineCreateOrUpdateBehavior.CalledWithInput.Pop().VM
+				Expect(vm.Properties.StorageProfile.ImageReference).ToNot(BeNil())
+				Expect(vm.Properties.StorageProfile.ImageReference.CommunityGalleryImageID).ToNot(BeNil())
+				parts := strings.Split(*vm.Properties.StorageProfile.ImageReference.CommunityGalleryImageID, "/")
+				Expect(parts[2]).To(Equal(expectedGalleryURL))
+				Expect(parts[4]).To(Equal(expectedImageDefinition))
+
+				// Need to reset env since we are doing these nested tests
+				cluster.Reset()
+				azureEnv.Reset()
+			},
+			Entry("Gen2, Gen1 instance type with AKSUbuntu image family",
+				"Standard_D2_v5", "Ubuntu2204", "2204gen2containerd", v1alpha2.AKSUbuntuPublicGalleryURL),
+			Entry("Gen1 instance type with AKSUbuntu image family",
+				"Standard_D2_v3", "Ubuntu2204", "2204containerd", v1alpha2.AKSUbuntuPublicGalleryURL),
+			Entry("ARM instance type with AKSUbuntu image family",
+				"Standard_D16plds_v5", "Ubuntu2204", "2204gen2arm64containerd", v1alpha2.AKSUbuntuPublicGalleryURL),
+			Entry("Gen2 instance type with AzureLinux image family",
+				"Standard_D2_v5", "AzureLinux", "V2gen2", v1alpha2.AKSAzureLinuxPublicGalleryURL),
+			Entry("Gen1 instance type with AzureLinux image family",
+				"Standard_D2_v3", "AzureLinux", "V2", v1alpha2.AKSAzureLinuxPublicGalleryURL),
+			Entry("ARM instance type with AzureLinux image family",
+				"Standard_D16plds_v5", "AzureLinux", "V2gen2arm64", v1alpha2.AKSAzureLinuxPublicGalleryURL),
+		)
+	})
 	Context("Instance Types", func() {
 		It("should support provisioning with no labels", func() {
 			ExpectApplied(ctx, env.Client, nodePool, nodeClass)
 			pod := coretest.UnschedulablePod(coretest.PodOptions{})
 			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, coreProvisioner, pod)
 			ExpectScheduled(ctx, env.Client, pod)
 		})
+		It("should support provisioning with Azure Linux", func() {
+			nodeClass.Spec.ImageFamily = lo.ToPtr(v1alpha2.AzureLinuxImageFamily)
+			ExpectApplied(ctx, env.Client, nodePool, nodeClass)
+			pod := coretest.UnschedulablePod(coretest.PodOptions{})
+			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, coreProvisioner, pod)
+			ExpectScheduled(ctx, env.Client, pod)
+
+			Expect(azureEnv.VirtualMachinesAPI.VirtualMachineCreateOrUpdateBehavior.CalledWithInput.Len()).To(Equal(1))
+			vm := azureEnv.VirtualMachinesAPI.VirtualMachineCreateOrUpdateBehavior.CalledWithInput.Pop().VM
+			Expect(vm.Properties.StorageProfile.ImageReference).ToNot(BeNil())
+			Expect(vm.Properties.StorageProfile.ImageReference.CommunityGalleryImageID).ToNot(BeNil())
+			Expect(strings.HasPrefix(*vm.Properties.StorageProfile.ImageReference.CommunityGalleryImageID, "/CommunityGalleries/AKSAzureLinux-f7c7cda5-1c9a-4bdc-a222-9614c968580b")).To(BeTrue())
```

Comment:
Can this use `v1alpha2.AKSAzureLinuxPublicGalleryURL`?

Reply:
Removing this test all together